firmware/library/L1_Drivers/ssp.hpp

```diff
@@ -246,27 +213,113 @@ class Ssp final : public SspInterface, protected Lpc40xxSystemController
     }
     return static_cast<uint16_t>(ssp_registers[pssp]->DR);
   }
-  // Sets the various modes for the Peripheral
-  // @param mode - master or slave mode
-  // @param frame - format for Peripheral data to use
-  // @param size - number of bits per frame
-  void SetPeripheralMode(MasterSlaveMode mode, FrameMode frame,
-                         DataSize size) override
+
+  /// Sets up SSP peripheral as SPI master
+  void SetSpiMasterDefault() override
   {
-    master_mode_  = mode;
-    frame_format_ = frame;
-    data_size_    = size;
+    uint32_t pssp = static_cast<uint32_t>(pssp_);
+
+    // first clear the appropriate register locations
+    ssp_registers[pssp]->CR0 &=
+        ~((0xF << kDataBit) | (0x3 << kFrameBit) | (0x3 << kPolarityBit) |
+        (0xFF << kDividerBit));
+    ssp_registers[pssp]->CR1 &= ~(1 << kMasterModeBit);
+    ssp_registers[pssp]->CPSR &= ~(0xFF);
+
+    // set master mode, 8-bit datasize, and SPI frame
+    ssp_registers[pssp]->CR1 |= (kMaster << kMasterModeBit);
+    ssp_registers[pssp]->CR0 |= (kEight << kDataBit) | (kSpi << kFrameBit);
+
+    // Set clk polarity, clk phase, and SCR divider
+    ssp_registers[pssp]->CR0 |= (0x1 << kPolarityBit) | (0x0 << kPhaseBit);
+    ssp_registers[pssp]->CR0 |= (0x0 << kDividerBit);
+    // Set prescaler
+    ssp_registers[pssp]->CPSR |= (0x2);
   }
 
-  // Sets the clock rate for the Peripheral
-  void SetClock(bool polarity, bool phase, uint8_t divider,
-                uint8_t prescaler) override
+  /// Sets the various modes for the Peripheral
+  /// @param mode - master or slave mode
+  /// @param frame - format for Peripheral data to use
+  /// @param size - number of bits per frame
+  void SetPeripheralMode(MasterSlaveMode mode,

```

Comment:
For this method and the rest that do bit manipulation, I believe that the best way forward for this control register is to create a bitfield overlay. See `L1/can.hpp` and how a bit field is used with the register.

Reply:
Oh, like the register mapping from CMPE146? I can only work on this in the evenings, what's our dead date?

Reply:
Yeah, same here. The due date was last weekend, but the absolute final time is this weekend since I will need to lock the repo as the semester starts. I don't want to require that students reinstall or pull from git in CmpE 30.